backend/app/core/bot.py

```diff
@@ -0,0 +1,117 @@
+import asyncio
+import json
+import os
+import sys
+from asyncio import exceptions
+from typing import Optional
+
+from nio import (
+    AsyncClient,
+    AsyncClientConfig,
+    InviteEvent,
+    LoginResponse,
+    MatrixRoom,
+    RoomMessageText,
+)
+
+from app.core.config import settings
+
+
+class BaseBotClient(AsyncClient):
+    def __init__(
+        self,
+        homeserver: str,
+        user: str = "",
+        device_id: Optional[str] = "",
+        store_path: Optional[str] = "",
+        config: Optional[AsyncClientConfig] = None,
+        ssl: Optional[bool] = None,
+        proxy: Optional[str] = None,
+    ):
+        super().__init__(homeserver, user, device_id, store_path, config, ssl, proxy)
+
+        self.password = None
+
+        # auto-join room invites
+        self.add_event_callback(self.cb_autojoin_room, InviteEvent)
+
+        # print all the messages we receive
+        self.add_event_callback(self.cb_print_messages, RoomMessageText)
+
+    async def login(self) -> None:
+        # Log in either using the session details file or the username, password (if the file dosen't exist)
+        if os.path.exists(settings.matrix_bot.session_details_file) and os.path.isfile(
+            settings.matrix_bot.session_details_file
+        ):
+            try:
+                with open(settings.matrix_bot.session_details_file, "r") as f:
+                    config = json.load(f)
+                    self.access_token = config["access_token"]
+                    self.user_id = config["user_id"]
+                    self.device_id = config["device_id"]
+
+            except IOError as err:
+                print(f"Couldn't load session from file. Logging in. Error: {err}")
+            except json.JSONDecodeError:
+                print("Couldn't read JSON file; overwriting")
+
+        # If the previous session is not stored, we'll log in with a password
+        if not self.user_id or not self.access_token or not self.device_id:
+            if self.password is None:
```

Comment:
I propose we just don't support login with a password at all. Just have an access token in the conifg file then we don't need to worry about making more device IDs etc.

Reply:
Done.